security-analysis/src/main/java/com/powsybl/security/Security.java

```diff
@@ -219,14 +332,105 @@ public static String printLimitsViolations(List<LimitViolation> violations, Limi
     }
 
     private static float getAbsValueLimit(LimitViolation violation) {
-        return Float.valueOf(Float.toString(violation.getLimit()) + (violation.getLimitReduction() != 1f ? " * " + violation.getLimitReduction() : ""));
+        return Math.abs(violation.getValue() - violation.getLimit() * violation.getLimitReduction());
     }
 
+    public static void printPreContingencyViolations(SecurityAnalysisResult result, Network network, Writer writer, TableFormatterFactory formatterFactory,
+                                                     LimitViolationFilter limitViolationFilter) {
+        printPreContingencyViolations(result, network, writer, formatterFactory, TableFormatterConfig.load(), limitViolationFilter);
+    }
+
+    public static void printPreContingencyViolations(SecurityAnalysisResult result, Network network, Writer writer, TableFormatterFactory formatterFactory,
+                                                     TableFormatterConfig formatterConfig, LimitViolationFilter limitViolationFilter) {
+        Objects.requireNonNull(result);
+        Objects.requireNonNull(writer);
+        Objects.requireNonNull(formatterFactory);
+        Objects.requireNonNull(formatterConfig);
+
+        NumberFormat numberFormat = getFormatter(formatterConfig.getLocale(), 4, 4);
+        NumberFormat percentageFormat = getFormatter(formatterConfig.getLocale(), 2, 2);
+
+        List<LimitViolation> filteredLimitViolations = limitViolationFilter != null
+                ? limitViolationFilter.apply(result.getPreContingencyResult().getLimitViolations(), network)
+                : result.getPreContingencyResult().getLimitViolations();
+
+        try (TableFormatter formatter = formatterFactory.create(writer,
+                "Pre-contingency violations",
+                formatterConfig,
+                new Column(ACTION),
+                new Column(COUNTRY),
+                new Column(BASE_VOLTAGE)
+                    .setHorizontalAlignment(HorizontalAlignment.RIGHT),
```

Comment:
The country and base_voltage columns are related to the voltage level: move the columns after the END column

Reply:
done